src/main/java/com/wwi21sebgroup5/cinema/components/CsvDataLoader.java

```diff
@@ -0,0 +1,41 @@
+package com.wwi21sebgroup5.cinema.components;
+
+import com.fasterxml.jackson.databind.MappingIterator;
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvSchema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.List;
+
+@Component
+public class CsvDataLoader {
+
+    private final Logger logger = LoggerFactory.getLogger(CsvDataLoader.class);
+
+    /**
+     *
+     * @param type Class type of the entries in the csv
+     * @param fileName Classpath to the csv resource
+     * @return Returns list of objects within the csv file parsed to the given type
+     * @param <T> Generic type to support multiple entities
+     */
+    public <T> List<T> loadObjectList(Class<T> type, String fileName) {
+        try {
+            CsvSchema bootstrapSchema = CsvSchema.emptySchema().withHeader();
+            CsvMapper mapper = new CsvMapper();
+            File file = new ClassPathResource(fileName).getFile();
+            MappingIterator<T> readValues =
+                    mapper.readerFor(type).with(bootstrapSchema).readValues(file);
+            return readValues.readAll();
+        } catch (Exception e) {
+            logger.error("Error occurred while loading object list from file " + fileName, e);
+            return Collections.emptyList();
+        }
+    }
+
```

Comment:
Was ist der Sinn dieser Klasse? Nur Testen oder spielt das auch im Produktiv System eine Rolle?

Reply:
Die Klasse dient für die generische Verarbeitung von csv-Dateiien in eine beliebe Datenklasse. Gerade wird sie nur für Städte benutzt, da werden bei Startup ca. 2000 Städte in die Datenbank geladen, stehen in cities.csv unter /ressources/static.
Kann aber auch für jegliche andere Daten benutzt werden.

Reply:
Don't mind me, während ich nicht in der Gruppe bin.
Aber wieso ladet ihr zum Anfang 2000 Städte in eure DB? Das ist ja ein performance Horror